### ISSUE TITLE
Filter not working?

### ISSUE THREAD

**Issue by eduardszoecs** [hrbrmstr/xmlview#5]
Useful package here!

However, the `add_filter` seems not to work:

```
library(xml2)
txt <- paste0("<note><to>Tove</to><from>Jani</from><heading>Reminder</heading>",
                               "<body>Don't forget me this weekend!</body></note>")
doc <- read_xml(txt)
xml_view(doc, add_filter = TRUE)
```

Every xpath that I enter in the Viewer (e.g. `.//to`) ends up in `undefined`.

```
> sessionInfo()
R version 3.2.3 (2015-12-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.3 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] xmlview_0.3.0 xml2_0.1.2   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.3       digest_0.6.9      R6_2.1.1          jsonlite_0.9.19   magrittr_1.5      httr_1.0.0        stringi_1.0-1     curl_0.9.4        devtools_1.9.1   
[10] tools_3.2.3       stringr_1.0.0     htmlwidgets_0.5.2 yaml_2.1.13       rsconnect_0.4.1.4 memoise_0.2.1     htmltools_0.3     knitr_1.12    
```

**Comment by timelyportfolio** [comment]
@EDiLD I can confirm it is not working in RStudio Viewer.  However, it does work in Chrome.  I'll dig a little deeper.

**Comment by hrbrmstr** [comment]
This is weird as it's working for me:

![image](https://cloud.githubusercontent.com/assets/509878/12325439/476898aa-ba98-11e5-8698-2c39b034176e.png)

**Comment by hrbrmstr** [comment]
oh, wait. linux. lemme fire up a VM later today and try it.

**Comment by eduardszoecs** [comment]
R Studio version: 

```
RStudio
Version 0.99.829 – © 2009-2015 RStudio, Inc.
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/538.1 (KHTML, like Gecko) RStudio Safari/538.1 Qt/5.4.0
```

**Comment by timelyportfolio** [comment]
It seems `DOMParser` is finicky.  `xml_view` does not work at all in Firefox.  I just get a blank box where the xpath should be and nothing else.

Here is my `sessionInfo()` for reference.

```
Session info --------------------------------------------------------------------
 setting  value                       
 version  R version 3.2.2 (2015-08-14)
 system   x86_64, mingw32             
 ui       RStudio (0.99.837)          
 language (EN)                        
 collate  English_United States.1252  
 tz       America/Chicago             
 date     2016-01-14                  

Packages ------------------------------------------------------------------------
 package    * version    date       source                             
 devtools     1.9.1.9000 2015-10-24 local                              
 digest       0.6.9      2016-01-08 CRAN (R 3.2.3)                     
 memoise      0.2.1      2014-04-22 CRAN (R 3.1.0)                     
 rstudio      0.99.95    2014-12-08 local                              
 rstudioapi   0.4.0      2016-01-04 Github (rstudio/rstudioapi@e115f4d)
```

**Comment by hrbrmstr** [comment]
Aye. It's toast in Firefox for now but I just pushed a release that makes it work in the RStudio Viewer on Linux desktops (tested on Ubuntu 14.04). Pls give it a go!

I feel compelled to add that I'm not going to be in a rush to get FFox support working either :-)

**Comment by hrbrmstr** [comment]
OK, it works with firefox now. It seems Firefox uses `textContent` vs `innerText`.

**Comment by eduardszoecs** [comment]
Works perfectly for me here! Thanks you for fixing this. really nice tool!

**Comment by timelyportfolio** [comment]
I can confirm that this works on Chrome, RStudio Viewer, and Firefox on my Windows 7 machine.  I did notice that the `R`-code button did not work in Firefox.  Might be another `innerText` issue.

**Comment by timelyportfolio** [comment]
Yep, this [line](https://github.com/hrbrmstr/xmlview/blob/master/inst/htmlwidgets/xmlview.js#L125) just has `innerText`.

**Comment by hrbrmstr** [comment]
danke. fixed in 0.4.1
